crates/net/common/src/metered_sender.rs

```diff
@@ -0,0 +1,64 @@
+//! Support for metering senders. Facilitates debugging by exposing metrics for number of messages
+//! sent, number of errors, etc.
+
+use metrics::Counter;
+use reth_metrics_derive::Metrics;
+use tokio::sync::mpsc::{
+    error::{SendError, TrySendError},
+    Sender,
+};
+
+/// Network throughput metrics
+#[derive(Metrics)]
+#[metrics(scope = "metered_sender")]
```

Comment:
Overall this PR looks good to me, although this highlights an issue with `metrics-derive` I think - `metered_sender` is not really a good name, it's way too generic. cc @rkrasiuk is there a way we can change this scope at the site it is used? My thinking here is that a metric like `reth.network.p2p.metered_sender.messages_sent` is not a super helpful name.

Reply:
Yeah, was a concern of mine as well - not super familiar with the metrics-derive crate but I can try playing around with some way to change the metric scope where it's used

Reply:
Also, if we want to use this in multiple places, we'd want some way to distinguish which channel's metrics we're looking at

Reply:
Oh lol all of this was mentioned in a separate comment thread my b